stdlib.sh

```diff
@@ -978,6 +978,37 @@ direnv_version() {
   "$direnv" version "$@"
 }
 
+# Usage: on_git_branch [<branch_name>]
+#
+# Returns 0 if within a git repository with the given branch name. If no branch
+# name is provided, then returns 0 when within _any_ branch. Requires the git
+# command to be installed.
+# Returns 1 otherwise.
+#
+# When a branch is specified, .git/HEAD is watched so that entering/exiting a
+# branch triggers a reload.
+#
+# Example:
+#
+#    if on_git_branch develop; then
+#      echo "Remember to merge with upstream regularly!"
+#    fi
+#
+#    if on_git_branch; then
+#      echo "Thanks for contributing to a GitHub project!"
+#    fi
+#
+on_git_branch() {
+  if ! has git; then return 1; fi
```

Comment:
how about also adding a message like `log_error "on_git_branch needs git, which was not found on your system"`

Reply:
Done.